lib/uplink/ext/test_utils.go

```diff
@@ -0,0 +1,176 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package main
+
+// #cgo CFLAGS: -g -Wall
+// #include <stdlib.h>
+// #ifndef STORJ_HEADERS
+//   #define STORJ_HEADERS
+//   #include "c/headers/main.h"
+// #endif
+import "C"
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"testing"
+	"unsafe"
+
+	"github.com/skyrings/skyring-common/tools/uuid"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap/zaptest"
+
+	"storj.io/storj/internal/testcontext"
+	"storj.io/storj/internal/testplanet"
+	"storj.io/storj/lib/uplink"
+	"storj.io/storj/pkg/macaroon"
+	"storj.io/storj/satellite/console"
+)
+
+/* C types */
+
+// Cchar is an alias to the c character pointer type
+type Cchar = *C.char
+type CUint = C.uint
+
+/* Ref types */
+
+// CAPIKeyRef is an alias to the APIKeyRef_t c type
+type CAPIKeyRef = C.APIKeyRef_t
+type CProjectRef = C.ProjectRef_t
+
+var (
+	CLibDir, CSrcDir, CTestsDir, LibuplinkSO string
+
+	testConfig = new(uplink.Config)
+)
+
+func init() {
+	// TODO: is there a cleaner way to do this?
+	_, thisFile, _, _ := runtime.Caller(0)
+	CLibDir = filepath.Join(filepath.Dir(thisFile), "c")
+	CSrcDir = filepath.Join(CLibDir, "src")
+	CTestsDir = filepath.Join(CLibDir, "tests")
+	LibuplinkSO = filepath.Join(CLibDir, "..", "uplink-cgo.so")
+
+	testConfig.Volatile.TLS.SkipPeerCAWhitelist = true
+}
+
+func runCTests(t *testing.T, ctx *testcontext.Context, envVars []string, srcGlobs ...string) {
+	srcGlobs = append([]string{
+		LibuplinkSO,
+		filepath.Join(CTestsDir, "unity.c"),
+		filepath.Join(CTestsDir, "helpers.c"),
+		filepath.Join(CSrcDir, "*.c"),
+	}, srcGlobs...)
+	testBinPath := ctx.CompileC(srcGlobs...)
+
+	if dir, ok := os.LookupEnv("STORJ_C_TEST_BIN_DIR"); ok {
```

Comment:
Can you add comment how this can be used?

Reply:
o right, yea!